common/test/acceptance/tests/lms/test_account_settings.py

```diff
@@ -180,6 +182,9 @@ def test_all_sections_and_fields_are_present(self):
             }
         ]
 
+        if not FEATURES.get('ENABLE_TIME_ZONE_PREFERENCE'):
+            del expected_sections_structure[0]['fields'][6]
```

Comment:
Instead of making this conditional in the tests, add a test for the flag being True and one for the flag being False,  controlling the setting using [`self.modify_settings()`](https://docs.djangoproject.com/en/1.9/topics/testing/tools/#django.test.SimpleTestCase.modify_settings).